src/read.c

```diff
@@ -5429,6 +5444,40 @@ avifResult avifDecoderReset(avifDecoder * decoder)
         decoder->image->transformFlags |= AVIF_TRANSFORM_IMIR;
         decoder->image->imir = imirProp->u.imir;
     }
+#if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
+    if (gainMapProperties != NULL) {
+        // The transformative properties of the gain map item have to match those of the primary item.
```

Comment:
Note: I am sorry this comment is verbose. The first sentence is the key point. Please feel free to address this issue in other ways; it is not necessary to accept my suggestion.

I still think we need to clarify whether this requirement comes from a (draft) standard or libavif imposes this requirement based on the justification in `avifRWStreamWriteProperties()`.

It is fine to refer to a draft standard because the gain map code is guarded by the `AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP` macro.

However, the proposed commit message implies this requirement is imposed by libavif. We can add the proposed commit message (slightly edited) as a comment here:

```
libavif requires the bitstream contain the same pasp, clap, irot, imir
properties for both the base and gain map image items used as input to
the tone-mapped derived image item. libavif also requires the tone-mapped
derived image item itself not be associated with these properties. This is
enforced at encoding. Other patterns are rejected at decoding.
```

This comment could also be added to line 4868.